src/wallet/wallet.cpp

```diff
@@ -5569,29 +5574,17 @@ int CMerkleTx::GetDepthInMainChain(const CBlockIndex* &pindexRet) const
     int nResult;
 
     if (hashUnset())
-        nResult = 0;
-    else {
-        AssertLockHeld(cs_main);
+        return 0;
```

Comment:
changes below are from yet another PR, not even from 11041 it seems

Reply:
Hmm, I agree these changes don't come from 11041... I can't find the PR it comes from... It looks like a good refactor. Can you find the PR it comes from?

Reply:
Found it: https://github.com/dashpay/dash/commit/f679e2431669d8fce270b2627b1908e1201f13e5#diff-1f2db0e4d5c12d109c7f0962333c245b49b696cb39ff432da048e9d6c08944d8R4056-R4086 tl;dr: it was modified to work with legacy IS "confirmations". This is no longer needed after https://github.com/dashpay/dash/commit/0a6f47323ba9da7a25e6bcb8b11aa4b21e6c71fd#diff-1f2db0e4d5c12d109c7f0962333c245b49b696cb39ff432da048e9d6c08944d8L5379-L5381, I think we can bring back the original logic here to reduce the amount of future merge conflicts. Could simply drop 6f1dbc1 OR squash it into df77902 and then add another commit on top which would revert Dash-specific code explicitly, I'm ok with either approach.